workitem/workitemtype_repository.go

```diff
@@ -44,6 +45,42 @@ func (r *GormWorkItemTypeRepository) LoadByID(ctx context.Context, id uuid.UUID)
 	return res, nil
 }
 
+// Exists returns true if the work item with the given ID exists
+
+// TODO(kwk): Implement tests for this and use more performant query. Currently
+// the logic is borrowed from the Load function except the space check which
+// MUST NOT be in here because without space templates we're creating WIs from
+// WITs out of the system space but in their own space.
+func (r *GormWorkItemTypeRepository) Exists(ctx context.Context, id uuid.UUID) (bool, error) {
+	log.Info(ctx, map[string]interface{}{
+		"wit_id": id,
+	}, "Checking if work item type exists")
+	found := false
+	res, ok := cache.Get(id)
+	if !ok {
+		log.Info(ctx, map[string]interface{}{
+			"wit_id": id,
+		}, "Work item type doesn't exist in the cache. Loading from DB...")
+		res = WorkItemType{}
+
+		db := r.db.Model(&res).Where("id=?", id).First(&res)
+		if db.RecordNotFound() {
+			log.Error(ctx, map[string]interface{}{
+				"wit_id": id,
+			}, "work item type not found")
+			return false, nil
+		}
+		if err := db.Error; err != nil {
+			return false, errors.NewInternalError(err.Error())
+		}
+		cache.Put(res)
+		found = true
+	} else {
+		found = true
```

Comment:
@hectorj2f fixed here: 23e0621 . I now use the `SELECT EXISTS` pattern in order to only hit the cache or at maximum the DB index when checking for the existence of a WIT. Maybe this pattern can be taken over to other places to avoid loading a complete object when a check for existence is enough.